src/main/java/org/sonar/cxx/checks/UsingNamespaceCheck.java

```diff
@@ -10,12 +10,16 @@
 import org.sonar.cxx.squidbridge.checks.SquidCheck;
 import org.sonar.cxx.tag.Tag;
 
+// In case you are adding a .html description in resources, the .html file name should match the rule key.
+// In this sample the name must be 'UsingNamespace.html'.
 @Rule(
   key = "UsingNamespace",
-  priority = Priority.BLOCKER,
-  name = "Using namespace directives are not allowed",
-  tags = {Tag.CONVENTION},
-  description = "Using namespace directives are not allowed.")
+   priority = Priority.BLOCKER,
+   name = "Using namespace directives are not allowed",
+   tags = {Tag.CONVENTION}
+// second possibility to add a rule description:
+//,description = "Using namespace directives are not allowed."
```

Comment:
The description here can be retained. If we use the addHtmlDescription method, we can override the description. If you do not use the addHtmlDescription  method, we can use this description as basic description.

Reply:
Unfortunately I haven't tried what happens when both are in? Does the html description overwrite the description?

Reply:
I tested this part of the code and it worked fine.
test version :
cxx: master (c753dd71932baabd7dcbe57e3a7985a48d733a60)
plugin: master (d221e457933cd45975fa56e28cc8d3a35cd0383c)

![image](https://github.com/SonarOpenCommunity/cxx-custom-checks-example-plugin/assets/137695008/80b59106-de48-467d-8b57-bded4252aec1)

This design idea comes from the Sonar official rules custom plug-in demo.
![image](https://github.com/SonarOpenCommunity/cxx-custom-checks-example-plugin/assets/137695008/de669ee2-1543-4551-affb-f9dd9473d38b)
![image](https://github.com/SonarOpenCommunity/cxx-custom-checks-example-plugin/assets/137695008/e574dc98-61a2-4813-9e07-cdd60d0d793a)